NextCloud.py

```diff
@@ -805,6 +807,97 @@ def delete_all_notifications(self):
 class UserLDAP(WithRequester):
     API_URL = "/ocs/v2.php/apps/user_ldap/api/v1/config"
 
+    # (name_stem, argname, docstring)
+    CONFIG_KEYS = [
+        "ldapHost",
+        "ldapPort",
+        "ldapBackupHost",
+        "ldapBackupPort",
+        "ldapBase",
+        "ldapBaseUsers",
+        "ldapBaseGroups",
+        "ldapAgentName",
+        "ldapAgentPassword",
+        "ldapTLS",
+        "turnOffCertCheck",
+        "ldapUserDisplayName",
+        "ldapGidNumber",
+        "ldapUserFilterObjectclass",
+        "ldapUserFilterGroups",
+        "ldapUserFilter",
+        "ldapUserFilterMode",
+        "ldapGroupFilter",
+        "ldapGroupFilterMode",
+        "ldapGroupFilterObjectclass",
+        "ldapGroupFilterGroups",
+        "ldapGroupMemberAssocAttr",
+        "ldapGroupDisplayName",
+        "ldapLoginFilter",
+        "ldapLoginFilterMode",
+        "ldapLoginFilterEmail",
+        "ldapLoginFilterUsername",
+        "ldapLoginFilterAttributes",
+        "ldapQuotaAttribute",
+        "ldapQuotaDefault",
+        "ldapEmailAttribute",
+        "ldapCacheTTL",
+        "ldapUuidUserAttribute",
+        "ldapUuidGroupAttribute",
+        "ldapOverrideMainServer",
+        "ldapConfigurationActive",
+        "ldapAttributesForUserSearch",
+        "ldapAttributesForGroupSearch",
+        "ldapExperiencedAdmin",
+        "homeFolderNamingRule",
+        "hasPagedResultSupport",
+        "hasMemberOfFilterSupport",
+        "useMemberOfToDetectMembership",
+        "ldapExpertUsernameAttr",
+        "ldapExpertUUIDUserAttr",
+        "ldapExpertUUIDGroupAttr",
+        "lastJpegPhotoLookup",
+        "ldapNestedGroups",
+        "ldapPagingSize",
+        "turnOnPasswordChange",
+        "ldapDynamicGroupMemberURL",
+        "ldapDefaultPPolicyDN",
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super(UserLDAP, self).__init__(*args, **kwargs)
+        for key in self.CONFIG_KEYS:
+            key_name = re.sub('ldap', '', key)
+            key_name = re.sub('([a-z0-9])([A-Z])', r'\1_\2', key_name).lower()
+
+            # create and add getter method
+            getter_name = "get_ldap_{}".format(key_name)
+            exec(
```

Comment:
I don't like the `exec` here. I propose to do something else - you probably want to have a generic (maybe private) getter/setter that accepts the `key` as well, and you somehow just stuff the class/instance with wrappers that behave in a way that e.g. `get_ldap_user_filter` calls `_get_ldap_property("user_filter")`.
You can use `lambda` to do the wrap, and you can even add those methods to the class itself (i.e. in the module-level code, not in the class constructor).

Reply:
Thanks for suggestions! Yes, using `exec` was bad idea. I refactored it: moved to module level and attaching methods to class and rewritten `exec` to functions. Although, I used def functions, not lambdas for explicity, is that ok ?
Also, I left `get_ldap_config`, `edit_ldap_config` not private and open for user in case if somebody will need to see full config view and make bulk update of keys. As well use them in config keys setters/getters.